pkg/spanconfig/spanconfigkvsubscriber/kvsubscriber.go

```diff
@@ -276,6 +306,8 @@ func (s *KVSubscriber) handlePartialUpdate(
 		// avoid this mutex.
 		s.mu.internal.Apply(ctx, false /* dryrun */, ev.(*bufferEvent).Update)
 	}
+	timeSinceLastUpdate := timeutil.Unix(0, ts.WallTime).Sub(timeutil.Unix(0, s.mu.lastUpdated.WallTime)).Nanoseconds()
```

Comment:
Worth not special casing this metric to just partial updates I think. Perhaps make a `setLastUpdatedLocked` helper, use it in both partial+complete update handling and update the gauge there. Also I think the following form is simpler + allocation efficient: 
```
s.metrics.NameOfField.Update(timeutil.Since(ts.GoTime()).Nanoseconds())
```